v2/internal/reconcilers/arm/azure_generic_arm_reconciler_instance.go

```diff
@@ -809,7 +810,7 @@ func (r *azureDeploymentReconcilerInstance) deleteResource(
 		return ctrl.Result{}, nil
 	}
 
-	err := checkSubscription(resourceID, r.ARMConnection.SubscriptionID()) // TODO: Possibly we should pass this in as a parameter?
+	err := r.checkSubscription(resourceID) // TODO: Possibly we should pass this in as a parameter?
```

Comment:
minor: Probably can remove this TODO? I am not sure what it was referencing but the code seems OK to me now